astropy/cosmology/funcs.py

```diff
@@ -253,3 +253,47 @@ def f(z):
         raise CosmologyError(f"Best guess z={res.x} is very close to the lower z limit {zmin}.\n"
                              "Try re-running with a different zmin.")
     return res.x
+
+
+def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500,
+               method='Brent', bracket=None, verbose=False):
+    """``scalar_z_at_value`` for vectorized inputs. Docstring set below."""
+    # `fval` can be a Quantity, which isn't (yet) compatible w/ `numpy.nditer`
+    # so we strip it of units for broadcasting and restore the units when
+    # passing the elements to `scalar_z_at_value`.
+    unit = getattr(fval, 'unit', 1)  # can be unitless
+
+    # bracket must be an object array (assumed to be correct) or a 'scalar'
+    # bracket: 2 or 3 elt sequence
+    if not isinstance(bracket, np.ndarray):  # 'scalar' bracket
+        if bracket is not None and len(bracket) not in (2, 3):
+            raise ValueError("`bracket` is not an object array "
+                             "nor a 2 (or 3) element sequence.")
+        else:  # munge bracket into a 1-elt object array
+            bracket = np.array([bracket, ()], dtype=object)[:1]
```

Comment:
I think your scalar may not work because you create an 1-d array here - could add `.squeeze()` to ensure you get shape `()`.

Reply:
Done! works.